feat(illustrations): NO-JIRA new illustrations added

fede-dp opened the pull request:
# feat(illustrations): NO-JIRA new illustrations added

<!--- Feel free to remove any unused sections -->

## Obligatory GIF (super important!)

<!--- go to giphy.com, pick a gif, click share, copy gif link, paste within the () brackets below. -->

![excited-jump-gif-by-moodman](https://github.com/user-attachments/assets/ae23dad5-3911-4dcc-9a44-e55298f536c6)

## :hammer_and_wrench: Type Of Change

<!--- Tick or place an `x` for the type of change. Should match the type in the commit message / PR title
in https://github.com/dialpad/dialtone/blob/staging/.github/COMMIT_CONVENTION.md -->

These types will increment the version number on release:

- [ ] Fix
- [x] Feature
- [ ] Performance Improvement
- [ ] Refactor

These types will not increment the version number, but will still deploy to documentation site on release:

- [ ] Documentation
- [ ] Build system
- [ ] CI
- [ ] Style (code style changes, not css changes)
- [ ] Test
- [ ] Other (chore)

## :book: Jira Ticket

<!--- Enter the URL of the Jira ticket associated with this PR -->

## :book: Description

Added new illustrations

<!--- Describe specifically what the changes are -->

## :bulb: Context

<!--- Describe the purpose of the changes -->
<!--- Why did we make these changes? -->
<!--- What problem(s) do they solve? -->

## :pencil: Checklist

<!--- Tick or place an `x` in all of the checkboxes that apply -->
<!--- Remove checkboxes that do not apply -->

For all PRs:

- [x] I have ensured no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).
- [x] I have reviewed my changes.
- [x] I have added all relevant documentation.
- [x] I have considered the performance impact of my change.

For all Vue changes:

- [ ] I have added / updated unit tests.
- [ ] I have made my changes in Vue 2 and Vue 3. Note: you may sync your changes from Vue 2 to Vue 3 (or vice versa) using the `./scripts/dialtone-vue-sync.sh` script. Read docs here: [Dialtone Vue Sync Script](../packages/dialtone-vue3/.github/CONTRIBUTING.md#dialtone-vue-sync-script)
- [ ] I have validated components with a screen reader.
- [ ] I have validated components keyboard navigation.

For all CSS changes:

- [ ] I have used design tokens whenever possible.
- [ ] I have considered how this change will behave on different screen sizes.
- [ ] I have visually validated my change in light and dark mode.
- [ ] I have used gap or flexbox properties for layout instead of margin whenever possible.

If new component:

<!--- There are lots of things to remember when adding a new component to the system! This is so you don't forget any of them. -->

- I am exporting any new components or constants:
  - [ ] from the index.js in the component directory.
  - [ ] from the index.js in the root (either `packages/dialtone-vue2` or `packages/dialtone-vue3`).
- [ ] I have added the styles for the new component to the `packages/dialtone-css` package.
- [ ] I have created a page for the new component on the documentation site in `apps/dialtone-documentation`.
- [ ] I have added the new component to `common/components_list.js`
- [ ] I have created a component story in storybook
- [ ] I have created story / stories for any relevant component variants in storybook
- [ ] I have created a docs page for the component in storybook.
- [ ] I have checked that changing all props/slots via the UI in storybook works as expected.

## :crystal_ball: Next Steps

<!--- Describe any future changes that need to be made after merging the PR, especially any follow up tasks after release. -->

## :camera: Screenshots / GIFs

<!--- Add these if necessary. Since we have deploy previews for every PR it may not always be. -->
<!--- Link any screenshots / GIFs below -->

## :link: Sources

<!--- Add any links to external reference material -->

braddialpad commented on gulpfile.js:
```diff
@@ -601,6 +607,7 @@ const transformStrokeToFill = function (done) {
           .transform(
               `${paths.svgs.newInputRoot}/${category}/`,
               `${paths.svgs.newOutputRoot}/${category}/`,
+              { traceResolution: 600, showProgressBar: true },
```
Do we need this anymore? We aren't transforming anymore right, should we just remove the library and all the code related to it or am I not understanding the process correctly?

juliodialpad replied:
We are going to use it to transform icons from `/newIcons` gitignored directory and save it to `/lib/build/svg/v7` I'll add the documentation

braddialpad replied:
You mentioned in this PR that Francis had already manually converted them via figma, is there still further conversion needed via this transform?

juliodialpad replied:
Yes, we were discussing with @francisrupert that there might be some icons we would like to convert using this technique as Figma has some bugs too.

braddialpad commented:
How would you actually import these tokens if you were using the package, would it be a subdir?

juliodialpad commented:
> How would you actually import these tokens if you were using the package, would it be a subdir?

Yes, right now is a subdir `@dialpad/dialtone/lib/dist/svg/v7/**/*.svg` That approach is ok or would it be better to export the `Vue` version of the icons?

juliodialpad commented on gulpfile.js:
```diff
@@ -637,11 +644,10 @@ const buildNewSVGIcons = function (done) {
-  return src(`${paths.svgs.newInputRoot}/**/*.svg`)
-  // replace any instances of the primary color in SVG with the theme class
+  return src(paths.svgs.version7Input)
     .pipe(addAttrsToSVG())
```
Yes, I'm adding that into https://github.com/dialpad/dialtone/pull/719

braddialpad commented:
> > How would you actually import these tokens if you were using the package, would it be a subdir?
> 
> Yes, right now is a subdir `@dialpad/dialtone/lib/dist/svg/v7/**/*.svg` That approach is ok or would it be better to export the `Vue` version of the icons?

hmmm we should probably do vue versions since that's how developers currently use the icons in product. Might be a bit confusing if they have to switch to SVG.

juliodialpad commented:
> > > How would you actually import these tokens if you were using the package, would it be a subdir?
> > 
> > 
> > Yes, right now is a subdir `@dialpad/dialtone/lib/dist/svg/v7/**/*.svg` That approach is ok or would it be better to export the `Vue` version of the icons?
> 
> hmmm we should probably do vue versions since that's how developers currently use the icons in product. Might be a bit confusing if they have to switch to SVG.

Vue implementation is added on https://github.com/dialpad/dialtone/pull/719 which has this branch as its base.

braddialpad commented:
Great thanks!

francisrupert reviewed (commented):
As far this specific PR goes, I have nothing further, especially with #719 being the real result. Remaining discussion appears to be Vue-centric, so I'll defer to @braddialpad for PR approval.

fede-dp committed:
new illustrations

fede-dp committed:
Implement structural updates and optimizations across multiple modules

francisrupert commented on packages/dialtone-icons/src/svg/illustrations/spot-illustrations/Theres-nothing-to-see-here.svg:
Rename this one to `nothing-to-see-here`. Is it not showing because of the many vector points... or is it because it was name `Theres...` with capital T instead of `theres...`?

francisrupert replied:
<img width="980" alt="image" src="https://github.com/user-attachments/assets/58e37cea-88fa-424b-9a0d-1b90f5abd888" />

jcarter-dp committed:
Update with new illustration and renaming illustration

Updates:
- New illustration called "Magnifier" added
- Renamed illustration from "Theres-nothing-to-see-here"
to "nothing-to-see-here"

jcarter-dp commented:
@francisrupert Got new illustration added and renamed the "nothing-to-see-here". Its also working fine

francisrupert reviewed (approved):
All good, thx! Merge away!
